bam2depth.c

```diff
@@ -712,16 +722,21 @@ static void usage_exit(FILE *fp, int exit_status)
     fprintf(fp, "  -b FILE      Use bed FILE for list of regions\n");
     fprintf(fp, "  -f FILE      Specify list of input BAM/SAM/CRAM filenames\n");
     fprintf(fp, "  -X           Use custom index files (in -X *.bam *.bam.bai order)\n");
-    fprintf(fp, "  -g INT       Remove specified flags from default flag filter\n");
-    fprintf(fp, "  -G INT       Add specified flags to the default flag filter\n");
+    fprintf(fp, "  -g INT       Remove specified flags from default filter-out flag list\n");
+    fprintf(fp, "  -G, --excl-flags FLAGS\n");
+    fprintf(fp, "               Add specified flags to the  default filter-out flag list\n");
+    fprintf(fp, "               [UNMAP,SECONDARY,QCFAIL,DUP]\n");
+    fprintf(fp, "      --incl-flags FLAGS\n");
+    fprintf(fp, "               Only inclue records with at least one the FLAGs present [0]\n");
```

Comment:
Oops.  I'll fix the typo.

The include vs incl abbreviation is specific to `view`.  Depth, mpileup, coverage and consensus all have `incl-flags` but not `include-flags`.  So I think going with the common trend is the right thing here.

Reply:
The long option additions to `view` are the most recent, so I suppose I was trying to introduce a trend of enabling the full spellings. (And hadn't got around to making a PR accepting both inter­change­ably in the other subcommands.)

But what I really wanted to introduce was samtools/htslib#1361, which would allow unifying all this as a single option instead of adding all these variants to the various subcommands. However the maintainers have carried on adding variant options instead. If there seemed to be interest in the unified approach, I would be happy to dust it off.

Reply:
That's because 1361 is draft.  When I asked how it was going you said it's still undergoing refactoring, so it's unfair to complain we haven't adopted those policies yet and "have carried on adding variant options instead".

Besides, I always viewed that PR as an additional nomenclature (not "instead") and all new commands would be supporting both old and new syntax for consistency.

Reply:
I said “seemed to be interest in [the alternative approach of a unified option]”, not “adopted”.

Certainly I proposed the unified approach as a more user-friendly and mnemonic alternative to further adding to the plethora of flag-filtering options and therefore not adding new variant options (while preserving the already existing ones, of course).

By approving PRs continuing to add to the plethora of flag-filtering options, the maintainers have implicitly signalled that they are not particularly interested in the proposed alternative user interface. Thanks for clarifying.

Reply:
The --incl-flags and --require-flags are entirely new to this subcommand, so this part at least was adding new features.  Given they're new options they obviously needed some name, and making them compatible with other existing options seemed logical rather than exacerbating things with yet another mnemonic.  Please do not take it as a signal that the unified approach is not desired, but as this arrived first there really was no question as to their naming.